core/state_transition.go

```diff
@@ -410,13 +398,13 @@ func (st *StateTransition) applyCreateValidatorTx(createValidator *staking.Creat
 	return nil
 }
 
-func (st *StateTransition) applyEditValidatorTx(
+func (st *StateTransition) verifyAndApplyEditValidatorTx(
 	editValidator *staking.EditValidator, blockNum *big.Int,
 ) error {
-	if !st.state.IsValidator(editValidator.ValidatorAddress) {
-		return errValidatorNotExist
+	err := VerifyEditValidatorStakingMsg(st.state, st.bc, editValidator)
```

Comment:
Same here. Can be VerifyAndEdit...... and return the edited validator object, so you don't have to do it again after.